### ISSUE TITLE
Request support for (See Attached YML) - NewOne Smartplug Dimmer - WF39M

### ISSUE THREAD

**Issue by fireheadman** [make-all/tuya-local#1362]
Please see attached YML file, Took advice of Beared Tinker and created the configuration and tested it.

# Log Message

```
No Logs
```

# Information about DPS mappings

```
8 DPs available, skipped the DP for timer as I was not interested in this function.
If this is desired, I could probably sift through all the devices to make this work.

```

zbisbtukzuiifmp6

# Information about how the device functions
Just basic dimmer module with selector for Last State, Light Type, Min/Max Brightness, and Timer (skipped)

**Comment by fireheadman** [comment]
```
homeassistant:/config/custom_components/tuya_local/devices# cat newone_dimmer_smartplug.yaml

name: New One Dimmer Smartplug
products:
  - id: zbisbtukzuiifmp6
    model: WF39M
    name: New One Light Dimmer
primary_entity:
  entity: light
  name: Dimmer
  dps:
    - id: 1
      type: boolean
      name: switch
    - id: 2
      name: brightness
      type: integer
      range:
        min: 10
        max: 1000
      mapping:
        - scale: 3.92
secondary_entities:
  - entity: number
    name: Min brightness
    category: config
    icon: "mdi:lightbulb-on-50"
    dps:
      - id: 3
        type: integer
        name: value
        range:
          min: 10
          max: 1000
        unit: "%"
        mapping:
          - scale: 10
  - entity: number
    name: Max brightness
    category: config
    icon: "mdi:lightbulb-on"
    dps:
      - id: 5
        type: integer
        name: value
        range:
          min: 10
          max: 1000
        unit: "%"
        mapping:
          - scale: 10
  - entity: select
    name: Light Type
    category: config
    dps:
      - id: 4
        type: string
        name: option
        mapping:
          - dps_val: led
            value: LED
          - dps_val: incandescent
            value: Incandescent
          - dps_val: halogen
            value: Halogen
  - entity: select
    name: Restore Power State
    icon: "mdi:toggle-switch"
    category: config
    dps:
      - id: 14
        name: option
        type: string
        mapping:
          - dps_val: memory
            value: Last State
          - dps_val: "on"
            value: "On"
          - dps_val: "off"
            value: "Off"
  - entity: select
    name: Indicator
    icon: "mdi:led-on"
    category: config
    dps:
      - id: 21
        type: string
        name: option
        mapping:
          - dps_val: none
            value: "Off"
          - dps_val: relay
            value: On/Off Status
          - dps_val: pos
            value: Switch Position
```

**Comment by fireheadman** [comment]
![image](https://github.com/make-all/tuya-local/assets/19847195/06af87dc-23dc-418d-8efb-695bef13a318)
![image](https://github.com/make-all/tuya-local/assets/19847195/c3045209-be2e-4848-88d4-8423cf51a24f)
![image](https://github.com/make-all/tuya-local/assets/19847195/6e8bb50d-ce78-43d3-8811-138f7bd120aa)
![image](https://github.com/make-all/tuya-local/assets/19847195/232e4a58-c11c-4c60-bff4-22a0cb164e84)

**Comment by fireheadman** [comment]
Everything was tested and works as expected....
Hoping you can add to the collection.

These aren't too bad to create.... starting to get the hang of it.

**Comment by fireheadman** [comment]
If needed:    

https://github.com/fireheadman/tuya-local-yamls/blob/main/newone_dimmer_smartplug.yaml

**Comment by make-all** [comment]
Duplicate of #1273 (released in 2023.12.2)
